content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+---
+linktitle: Unbounded Arrays in AZSL
+title: AZSL, Binding Rules For Unbounded Arrays
+description: There are limitations into how many and what kind of Unbounded Arrays can be declared. 
+weight: 100
+---
+AZSL supports declaration of Unbounded Arrays inside `ShaderResourceGroup`s, but there are limitations into how many and what kind of Unbounded Arrays can be declared. The shader compiler, **AZSLc**, makes sure those rules are not violated.  
+  
+Here is a summary of the rules and limitations per platform (DX12, Vulkan and Metal). '--use-spaces' and '--unique-idx' are command line arguments for **AZSLc** that are used for each platform.  
+| --use-spaces   | --unique-idx | platform | Results |
+|----------------|--------------|----------|---------|
+| OFF | OFF |                   | Only the last SRG can contain one last unbounded array for each:<br> t[], u[], b[], s[]|
+| OFF | ON  |                   | Only the last SRG can contain one, and only one, last unbounded array across all resource types:<br>Only one of t[], u[], b[], s[]|
+| ON  | OFF | DX12              | Each SRG can contain one last unbounded array for each: <br> t[], u[], b[], s[]|
+| ON  | ON  | Vulkan<br> Metal  | Each SRG can contain one, and only one, last unbounded array across all resource types: <br> Only one of t[], u[], b[], s[]|
+  
+## Things to know before going over some examples:
+* "Register Space" in HLSL is the same as "Descriptor Set" in Vulkan.
+* When "--use-spaces" is used, **AZSLc** will assign a unique Register Space per SRG.
+* When "--unique-idx" is used, **AZSLc** will emit all resource descriptors in a Register Space to be in numerical sequence, regardless of the descriptor type. Platforms like Vulkan and Metal require this.
+* Unbounded arrays take over the whole range of register indices from the point they are being declared. This means two things:
+    1. It is possible to declare any kind of resources or any amount of bounded arrays of resources, BEFORE any given Unbounded Array of a particular kind.
+    2. Any resource declared AFTER an Unbounded Array will be considered an error.
+  
+## Examples
+All the examples shown here assume '--use-spaces' is ON when compiling the shader. Which means each SRG will be assigned to its own Register Space.
+
+### Example 1
+In this example there's only one SRG, and no unbounded arrays yet. It is useful to understand register space & index binding assignment.  
```

Comment:
```suggestion
### Example: Register assignments

This example demonstrates how index bindings are assigned to register spaces. It contains one SRG and no unbounded arrays.
```
* Give a more descriptive "Example" title. 
* Swap first and second sentence, so the first sentence explains what the example is about. Then, reword sentence to active voice.

Reply:
Fixed.